oci-labs/check-ins
# Feature 1257/delete unused ad hoc templates on request flow

To test this, start up the backend and go to:
```
localhost:3000/feedback/request/?for=1b4f99da-ef70-4a76-9b37-8bb783b749ad
```

Create an ad-hoc template and verify it appears on the page. You can create as many of these templates as you like. Then, select some recipients and dates and submit the feedback request. This should soft-delete any ad-hoc templates you created. So if you go back to `/feedback/request/?for=1b4f99da-ef70-4a76-9b37-8bb783b749ad`, those templates should no longer be visible.

In order to perform a soft-delete, there is a new endpoint in `FeedbackTemplateController`. A template can be soft-deleted by its ID via performing `DELETE` on the route `/services/feedback/{id}`. In order to delete *all* ad-hoc templates by a specific creator (as is done when submitting a feedback request), a separate route is used: `/services/feedback/creator/{creatorId}`. If there are any suggestions for changing how this route is used, please let us know.

Closes #1257
